fair/src/autogluon/fair/utils/group_metrics.py

```diff
@@ -0,0 +1,240 @@
+        val = self.func(*array)
+        return val.min(-1)
+
+
+class GroupDiff(BaseGroupMetric):
+    "helper class for reporting maximal difference in score between any pair of groups"
+
+    def __call__(self, *args):
+        array = self.build_array(args)
+        val = self.func(*array)
+        return val.max(-1) - val.min(-1)
+
+class GroupAvDiff(BaseGroupMetric):
+    "helper class for reporting average difference in score between all pairs of groups"
+
+    def __call__(self, *args):
+        array = self.build_array(args)
+        val = self.func(*array)
+        broadcast=(val[:,np.newaxis,:]-val[:,:,np.newaxis])
+        trunc=np.maximum(broadcast,0)
+        collate=trunc.sum(1).sum(1)/(val.shape[1]*(val.shape[1]-1)/2)
+        return collate
+
+
+class GroupRatio(BaseGroupMetric):
+    "helper class for reporting minimal score ratio  between any pair of groups"
+
+    def __call__(self, *args):
+        array = self.build_array(args)
+        val = self.func(*array)
+        return val.min(-1) / (1e-6 + val.max(-1))
+
+
+class Overall(BaseGroupMetric):
+    "helper class for reporting score over entire dataset"
+
+    def __call__(self, *args):
+        TP, FP, FN, TN = self.build_array(args)
+        val = self.func(TP.sum(1), FP.sum(1), FN.sum(1), TN.sum(1))
+        return val
+
+
+class GroupAverage(BaseGroupMetric):
+    "helper class for reporting scores averaged over groups"
+
+    def __call__(self, *args):
+        array = self.build_array(args)
+        val = self.func(*array)
+        return val.mean(-1)
+
+
+class GroupMetric(BaseGroupMetric):
+    """Broadcastable metrics used by efficient compute.
+        All methods either takes a single 3d numpy array as input or three vectors, y_true, y_pred, and groups
+        The matrix passed to any function is assumed to be of size
+        4 x entries x groups.
+        The first entry of the first axis corresponds to the number of True Negatives,
+        second False Negatives,
+        third False Positives, and
+        fourth True Positives.
+    """
+
+    def __init__(self, func, name, greater_is_better=True):
+        super().__init__(func, name, greater_is_better)
+        self.max = GroupMax(func, 'Maximal Group ' + name, greater_is_better=greater_is_better)
+        self.min = GroupMin(func, 'Minimal Group ' + name, greater_is_better=greater_is_better)
+        self.overall = Overall(func, 'Overall ' + name, greater_is_better=greater_is_better)
+        self.average = GroupAverage(func, 'Average Group ' + name, greater_is_better=greater_is_better)
+        self.diff = GroupDiff(func, 'Maximal Group Difference in ' + name, greater_is_better=False)
+        self.av_diff = GroupAvDiff(func, 'Average Group Difference in ' + name, greater_is_better=False)
+        self.ratio = GroupRatio(func, 'Minimal Group Ratio in ' + name, greater_is_better=True)
+        self.per_group = PerGroup(func, 'Per Group ' + name, greater_is_better=greater_is_better)
```

Comment:
Because they also require name strings and individual greater_is_better bools. 
The bool is particularly important as it indicates if we are minimizing or maximizing each objective.